hypothesis-python/src/hypothesis/internal/conjecture/providers.py

```diff
@@ -104,10 +111,11 @@
 #: Though, as ``backend="hypothesis"`` is the default setting, the above would
 #: typically not have any effect.
 #:
-#: The purpose of mapping to an absolute importable path, rather than the actual
-#: |PrimitiveProvider| class, is to avoid slowing down Hypothesis startup times
-#: by only importing alternative backends when required.
-AVAILABLE_PROVIDERS = {
+#: If you maintain a third-party backend, we strongly recommend providing an
+#: absolute importable path, rather than a |PrimitiveProvider| class. Hypothesis
+#: will defer importing class paths until required, which improves startup times
+#: for tests which don't request the backend.
```

Comment:
> I already wrote such a paragraph yep, encouraging string by default

Nice!  I think it's worth expanding on this though: "We strongly encourage plugin authors to ensure that `import hypothesis` does not automatically import your package, e.g. by registering a standalone `_mypkg_hypothesis_plugin` module rather than `mypgk._hypothesis_plugin`."

Reply:
I think the plugin example is confusing - it can be in a subdir as long as you're careful about imports - so I just said as much more explicitly